state_machine/applications/system-check/tests/current_test.py

```diff
@@ -0,0 +1,49 @@
+from lib.pycubed import cubesat
+import time
+
+def current_test():
+    measurements = 100
+    vals = [0] * measurements
+    outlier_count = 0
+    for i in range(measurements):
+        vals[i] = cubesat.battery_current
+        time.sleep(0.05)
+    sum = 0
+    for val in vals:
+        sum += val
+    avg = sum / 100
+    for val in vals:
+        if not ((avg - 2) <= val <= (avg + 2)):
+            outlier_count += 1
+
+    # if greater than 10% of the data points are far away from the average
+    # the sensor is giving strange and possibly inaccurate readings.
+    if (outlier_count >= measurements / 10):
+        return (f"current was inconsistent: {outlier_count} outliers", False)
+
+    # if you want to test on average current value that can be done here
+    # if not (-0.5 <= avg <= 0.5):
+    #     return (f"current did not return expected value: measurement {avg} mA", False)
+
+    return ("passed current test", True)
```

Comment:
I like to include data in the return string when I can. `avg` would be nice to include here (make sure to have units too).